lighthouse-core/audits/unsized-images.js

```diff
@@ -0,0 +1,127 @@
+ */
+'use strict';
+
+const Audit = require('./audit.js');
+const i18n = require('./../lib/i18n/i18n.js');
+const URL = require('./../lib/url-shim.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on whether all images have explicit width and height. This descriptive title is shown to users when every image has explicit width and height */
+  title: 'Image elements have explicit `width` and `height`',
+  /** Title of a Lighthouse audit that provides detail on whether all images have explicit width and height. This descriptive title is shown to users when one or more images does not have explicit width and height */
+  failureTitle: 'Image elements do not have explicit `width` and `height`',
+  /** Description of a Lighthouse audit that tells the user why they should include explicit width and height for all images. This is displayed after a user expands the section to see more. No character length limits. 'Learn More' becomes link text to additional documentation. */
+  description: 'Always include explicit width and height on your image elements to reduce layout shifting and improve CLS. [Learn more](https://web.dev/optimize-cls/#images-without-dimensions)',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+/**
+ * @fileoverview
+ * Audit that checks whether all images have width and height attributes.
+ */
+
+class SizedImages extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'unsized-images',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['ImageElements'],
+    };
+  }
+
+  /**
+   * @param {string} attr
+   * @return {boolean}
+   */
+  static isValidAttr(attr) {
+    // an img size attribute is valid for preventing CLS
+    // if it is a non-negative, non-zero integer
+    const NON_NEGATIVE_INT_REGEX = /^\d+$/;
+    const ZERO_REGEX = /^0+$/;
+    return NON_NEGATIVE_INT_REGEX.test(attr) && !ZERO_REGEX.test(attr);
+  }
+
+  /**
+   * @param {string} property
+   * @return {boolean}
+   */
+  static isValidCss(property) {
+    // an img css size property is valid for preventing CLS
+    // if it is defined, not empty, and not equal to 'auto'
+    // `undefined` and `''` are implicitly rejected as invalid
+    // because of their falsy short-circuit of && in isUnsizedImage
+    if (!property) return false;
+    return property !== 'auto';
+  }
+
+  /**
+   * @param {LH.Artifacts.ImageElement} image
+   * @return {boolean}
+   */
+  static isUnsizedImage(image) {
+    // images are considered sized if they have defined & valid values
```

Comment:
nit: re: all comments in this PR

our style is to format comments like this:

```
// Capitalize first letter. End with period.
```

Reply:
I will fix that!